phpBB/phpbb/di/container_builder.php

```diff
@@ -125,62 +125,88 @@ function __construct($phpbb_root_path, $php_ext)
 
-			$this->container = $this->create_container($this->container_extensions);
+				if ($this->use_extensions)
+				{
+					$this->load_extensions();
+				}
 
-			// Easy collections through tags
-			$this->container->addCompilerPass(new pass\collection_pass());
+				// Inject the config
+				if ($this->config_php_file)
+				{
+					$this->container_extensions[] = new extension\config($this->config_php_file);
+				}
 
-			// Event listeners "phpBB style"
-			$this->container->addCompilerPass(new RegisterListenersPass('dispatcher', 'event.listener_listener', 'event.listener'));
+				$this->container = $this->create_container($this->container_extensions);
 
-			// Event listeners "Symfony style"
-			$this->container->addCompilerPass(new RegisterListenersPass('dispatcher'));
+				// Easy collections through tags
+				$this->container->addCompilerPass(new pass\collection_pass());
 
-			$filesystem = new filesystem();
-			$loader     = new YamlFileLoader($this->container, new FileLocator($filesystem->realpath($this->get_config_path())));
-			$loader->load($this->container->getParameter('core.environment') . '/config.yml');
+				// Event listeners "phpBB style"
+				$this->container->addCompilerPass(new RegisterListenersPass('dispatcher', 'event.listener_listener', 'event.listener'));
 
-			$this->inject_custom_parameters();
+				// Event listeners "Symfony style"
+				$this->container->addCompilerPass(new RegisterListenersPass('dispatcher'));
 
-			if ($this->compile_container)
-			{
-				$this->container->compile();
+				$filesystem = new filesystem();
+				$loader     = new YamlFileLoader($this->container, new FileLocator($filesystem->realpath($this->get_config_path())));
+				$loader->load($this->container->getParameter('core.environment') . '/config.yml');
 
-				if ($this->use_cache)
+				$this->inject_custom_parameters();
+
+				if ($this->compile_container)
 				{
-					$this->dump_container($config_cache);
+					$this->container->compile();
+
+					if ($this->use_cache)
+					{
+						$this->dump_container($config_cache);
+					}
 				}
 			}
-		}
 
-		if ($this->compile_container && $this->config_php_file)
-		{
-			$this->container->set('config.php', $this->config_php_file);
+			if ($this->compile_container && $this->config_php_file)
+			{
+				$this->container->set('config.php', $this->config_php_file);
+			}
+
+			return $this->container;
 		}
+		catch (\Exception $e)
+		{
+			// Don't try to recover if we are in the development environment
+			if ($this->get_environment() === 'development')
+			{
+				throw $e;
+			}
 
-		return $this->container;
+			try
+			{
+				return $this
+					->without_extensions()
+					->without_cache()
+					->with_custom_parameters(array_merge($this->custom_parameters, [
+						'container_exception' => $e,
+					]))
+					->get_container();
+			}
+			catch (\Exception $_)
+			{
+				// Rethrow the original exception if it's still failing
+				throw $e;
+			}
```

Comment:
If the container is rebuilt and still throws an exception you won't get here. Instead it will try to build the container again and again.
You should tell the container builder after the first try that there already was a first try and it should not attempt to rebuild the container but instead throw the exception.